src/Geometry/Primitives/NonCSG/Rectangle.jl

```diff
@@ -103,7 +112,20 @@ function surfaceintersection(rect::Rectangle{T}, r::AbstractRay{T,3}) where {T<:
     end
 end
 
+"""returns the 2D vertices in the plane of the rectangle"""
+vertices(r::Rectangle{T},::Int = 0) where{T<:Real} = SMatrix{2,4}(vertices3d(r)[1:2,:])
+  
+
+"""returns the vertices of the rectangle in 3D"""
+vertices3d(r::Rectangle{T},::Int = 0) where{T<:Real} = SMatrix{3,4}(
+    point(r, -one(T), -one(T))...,
+    point(r, -one(T), one(T))...,
+    point(r, one(T), one(T))...,
+    point(r, one(T), -one(T))...)
```

Comment:
I'll benchmark it but I think it works out okay. point returns an SVector and the compiler usually is smarter in this case because it knows the size of the return value.

Reply:
it did allocate. But the latest version eliminates allocations for Ellipse, Hexagon, Rect for the vertices3d function.